evennia/contrib/tests.py

```diff
@@ -1717,6 +1717,710 @@ def test_generate(self):
+        dull_axe = create_object(
+                self.object_typeclass,
+                key='dull axe', location=self.char1.location)
+        timber = create_object(
+                self.object_typeclass,
+                key='timber', location=self.char1.location)
+        log = create_object(
+                self.object_typeclass,
+                key='log', location=self.char1.location)
+        parts = ['tree', 'axe']
+        results = (['sweat'] * 10) + ['dull axe'] + (['timber'] * 20) + (['log'] * 50)
+        recipe_dbref = self._good_recipe(
+            'lumberjack',
+            parts, results,
+            and_destroy_it=False
+        )
+
+        _destroy_objs_in_room(set(parts + results))
+
+        sps = sorted(parts)
+        expected = {key: len(list(grp)) for key, grp in itertools.groupby(sps)}
+        expected.update({r: 0 for r in set(results)})
+
+        self._arm(recipe_dbref, 'lumberjack', parts)
+        self._check_room_contents(expected)
+
+        self._use(','.join(parts), 'You are a Genius')
+        srs = sorted(set(results))
+        expected = {(key, len(list(grp))) for key, grp in itertools.groupby(srs)}
+        expected.update({p: 0 for p in set(parts)})
+        self._check_room_contents(expected)
+
+        # parts also appear in results
+        # causing a new puzzle to be armed 'automatically'
+        # i.e. the puzzle is self-sustaining
+        hole = create_object(
+                self.object_typeclass,
+                key='hole', location=self.char1.location)
+        shovel = create_object(
+                self.object_typeclass,
+                key='shovel', location=self.char1.location)
+        dirt = create_object(
+                self.object_typeclass,
+                key='dirt', location=self.char1.location)
+
+        parts = ['shovel', 'hole']
+        results = ['dirt', 'hole', 'shovel']
+        recipe_dbref = self._good_recipe(
+            'digger',
+            parts, results,
+            and_destroy_it=False,
+            expected_count=2
+        )
+
+        _destroy_objs_in_room(set(parts + results))
+
+        nresolutions = 0
+
+        sps = sorted(set(parts))
+        expected = {key: len(list(grp)) for key, grp in itertools.groupby(sps)}
+        expected.update({'dirt': nresolutions})
+
+        self._arm(recipe_dbref, 'digger', parts)
+        self._check_room_contents(expected)
+
+        for i in range(10):
+            self._use(','.join(parts), 'You are a Genius')
+            nresolutions += 1
+            expected.update({'dirt': nresolutions})
+            self._check_room_contents(expected)
+
+        # Uppercase puzzle name
+        balloon = create_object(
+                self.object_typeclass,
+                key='Balloon', location=self.char1.location)
+        parts = ['Balloon']
+        results = ['Balloon']
+        recipe_dbref = self._good_recipe(
+            'boom!!!',  # FIXME: uppercase name fails
```

Comment:
This is still outstanding. 
Uppercase letters cause failures. Needs investigation but seems like outside puzzles modules and inherent to search.

Reply:
Thanks @Griatch!

I removed the `FIXME:` as it's case-insensitive is the expected behavior.